product-service/functions/getProducts.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+const AWS = require("aws-sdk");
+const db = new AWS.DynamoDB.DocumentClient();
+
+const productTable = process.env.PRODUCT_TABLE;
+const stockTable = process.env.STOCK_TABLE;
+
+module.exports.handler = async (event) => {
+  const products = await db
+    .scan({
+      TableName: productTable,
+    })
+    .promise();
+
+  const stocks = await db
+    .scan({
+      TableName: stockTable,
+    })
+    .promise();
+
+  const stockMap = {};
+  stocks.Items.forEach((stock) => {
+    stockMap[stock.product_id] = stock;
+  });
+
+  const finalProducts = products.Items.map((product) => {
+    const stockInfo = stockMap[product.id] || { count: 0 };
+    return {
```

Comment:
Always return not found or other status as soon as possible to avoid compute in cloud. In this case you could have returned this at the beginning in case the product alone was not found